compiler/src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -1044,7 +1045,16 @@ object Erasure {
 
     override def typedClassDef(cdef: untpd.TypeDef, cls: ClassSymbol)(using Context): Tree =
       if cls.is(Flags.Erased) then erasedDef(cls)
-      else super.typedClassDef(cdef, cls)
+      else
+        val typedTree@TypeDef(name, impl @ Template(constr, _, self, _)) = super.typedClassDef(cdef, cls): @unchecked
+        // In the case where a trait extends a class, we need to strip any non trait class from the signature
+        // and accept the first one
+        val newParents = impl.parents.head :: impl.parents.tail.filter:
+          case Trees.Block(_, expr) =>
+            // Specific management for trait constructors (See #9216)
```

Comment:
I'm more in favour of mentioning the PR that added the specific management instead of a file. One can have more details by looking at the PR instead of the test.

Reply:
We always mention the test. This way we can reproduce the issue easily. I f you mention the PR we then need to go to the PR and figure out which one of the tests was the important one. Furthermore, it is possible that the behavior you mention in the comment does not manifest in a future version in the same way, in which case it would be impossible to figure out why this branch was added.